parsec/core/fs/storage/manifest_storage.py

```diff
@@ -88,6 +92,47 @@ async def _create_db(self):
                 );
                 """
             )
+            # Singleton storing the pattern_filter
+            cursor.execute(
+                """
+                CREATE TABLE IF NOT EXISTS pattern_filter
+                (
+                  _id INTEGER PRIMARY KEY NOT NULL,
+                  pattern TEXT NOT NULL,
+                  fully_applied INTEGER NOT NULL  -- Boolean
+                );
+                """
+            )
+            # Set the default pattern filter if it doesn't exist
+            cursor.execute(
+                """
+                INSERT OR IGNORE INTO pattern_filter(_id, pattern, fully_applied)
+                VALUES (0, ?, 0)""",
+                (EMPTY_PATTERN,),
+            )
+
+    # Pattern filter operations
+
+    async def get_pattern_filter(self) -> Tuple[Pattern, bool]:
+        async with self._open_cursor() as cursor:
+            cursor.execute("SELECT pattern, fully_applied FROM pattern_filter WHERE _id = 0")
+            reply = cursor.fetchone()
+            pattern, fully_applied = reply
+            return (re.compile(pattern), bool(fully_applied))
+
+    async def set_pattern_filter(self, pattern: Pattern):
+        async with self._open_cursor() as cursor:
+            cursor.execute(
+                """UPDATE pattern_filter SET pattern = ?, fully_applied = 0 WHERE _id = 0 AND pattern != ?""",
+                (pattern.pattern, pattern.pattern),
+            )
+
+    async def set_pattern_filter_fully_applied(self, pattern):
```

Comment:
wouldn't it be simpler to just have `async def set_pattern_filter(self, pattern: Pattern, fully_applied: bool=False):` ?

EDIT: after wondering about concurrency issue, I think I understand better the logic here: we have a two steps mechanism
- declare the pattern as changed by calling `set_pattern_filter()`
- then crawl the data to apply it, this is outside of the scope of this api
- finally mark the pattern as fully applied by calling `set_pattern_filter_fully_applied()`, the pattern must be provided here to make sure it hasn't been changed in the meantime (by calling `set_pattern_filter()` again)

If those assumptions are true, I think we should rename a bit this:
`change_pattern_filter()`
`mark_pattern_filter_fully_applied()`
and document how the two must be used together (and how concurrency supported handled)